app/models/transaction.rb

```diff
@@ -26,7 +28,10 @@ class Transaction < ActiveRecord::Base
   delegate :income?, :expense?, to: :category, allow_nil: true
 
   default_scope { order(created_at: :desc) }
-  scope :by_currency, ->(currency) { joins(:bank_account).where('bank_accounts.currency' => currency) }
+  scope :by_currency, ->(currency) { joins("INNER JOIN bank_accounts bank_account_transactions
+      ON bank_account_transactions.id = transactions.bank_account_id
+      AND bank_account_transactions.deleted_at IS NULL
+      AND bank_accounts.currency = '#{currency}'") }
```

Comment:
@Gataniel escape `currency`. Not sure why do you move it to `joins` instead `where`